### DISCUSSION TITLE
How should plugins subscribe to events?

### DISCUSSION THREAD

**Discussion by kizivat** [ianstormtaylor/slate#4265] [Q&A]
I was wondering how should plugins subscribe to events? 

I tried looking for an answer in `slate-history`'s source code as it apparently handles <kbd>Ctrl</kbd>+<kbd>Z</kbd> shortcut, but I couldn't find out how it does that.

**Comment by TheSpyder** [comment]
What sort of events do you mean? The undo shortcut is a cheat, the core knows history exists and just calls the undo/redo methods. (there's a bug at the moment where part of `slate-history` is bundled to achieve this)

https://github.com/ianstormtaylor/slate/blob/623960a7d14103b8cebe667019b4de5f8ad1fd61/packages/slate-react/src/components/editable.tsx#L886-L894

**Reply by kizivat** [reply]
I meant "DOM" events like `onClick`, `onKeyDown`, `onBlur` etc. on the `Editable` component. I ended up exporting the event handler from my plugin's module. Plugin's users need to import the event handler and use it on the `Editable` component. Example:

```js
import { withMyPlugin, keyDownHandler } from 'my-slate-plugin';
// ...
<Editable onKeyDown={keyDownHandler} />
// ...
```

I hope it makes sense, it's a while ago.
